### PR TITLE
Add org.js.nuclear.Nuclear

### PR DESCRIPTION
**Pull request by advaithm**
[nuclear]( https://nuclear.js.org) is a free music streaming program that pulls content from free sources all over the internet.

If you know mps-youtube, this is a similar music player but with a GUI. It's also focusing more on audio. Imagine Spotify which you don't have to pay for and with a bigger library.

### CODE REVIEW

**Comment by hfiguiere** [comment]
the manifest file name and the app-id don't match.

Also I doubt the app-id is correct.

**Comment by hfiguiere** [comment]
`org.js.nuclear.Nuclear` is IMHO more appropriate.

**Comment by advaithm** [comment]
oh sorry about the typos. i'll change the name to the one you suggested @hfiguiere

**Comment by hfiguiere** [comment]
bot, build org.js,nuclear.Nuclear

File: org.js.nuclear.Nuclear.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
+{
+    "app-id": "org.js,nuclear.Nuclear",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:2]: you put a comma instead of a period.

Comment (advaithm) [reply org.js.nuclear.Nuclear.json:2]: yeah sorry its pretty late at night for me.

**Comment by advaithm** [comment]
@hfiguiere i am very sorry about the typos. i should have checked for them.

**Comment by advaithm** [comment]
@hfiguiere  i fixed the code you can run the test again

**Comment by LeandroStanger** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
i figured out what was wrong. also Nuclear currently only supports x86_64.

File: org.js.nuclear.Nuclear.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    "base": "io.atom.electron.BaseApp",
+    "command": "run.sh",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--filesystem=host"
+    ],
+    "build-options" : {
+	"build-args": [ "--share=network" ],
```
Comment (bilelmoussaoui) [inline org.js.nuclear.Nuclear.json:16]: You can't have network access at build time as the builders are sandboxed.

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

File: org.js.nuclear.Nuclear.json (hunk 2 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json:16]: oh okay

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by hfiguiere** [comment]
> i figured out what was wrong. also Nuclear currently only supports x86_64.

If that's the case, add a `flathub.json` with the following content:

```json
{
    "only-arches": ["x86_64"]
}
```

**Comment by bilelmoussaoui** [comment]
bot, build org.js.nuclear.Nuclear

File: org.js.nuclear.Nuclear.json [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    "command": "run.sh",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--filesystem=host"
+    ],
+    "build-options" : {
+        "cflags": "-O2 -g",
+        "cxxflags": "-O2 -g",
```
Comment (bilelmoussaoui) [inline org.js.nuclear.Nuclear.json:17]:
```suggestion
```
already the default

File: org.js.nuclear.Nuclear.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    "runtime-version": "19.08",
+    "branch": "19.08",
+    "sdk": "org.freedesktop.Sdk",
+    "base": "io.atom.electron.BaseApp",
+    "command": "run.sh",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--filesystem=host"
```
Comment (bilelmoussaoui) [inline org.js.nuclear.Nuclear.json:13]: What does it need host filesystem access for?

Comment (advaithm) [reply org.js.nuclear.Nuclear.json:13]: It's for downloads of songs. You can set the download location and the locations of where songs are read from

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
based on some testing with some of the devs of nuclear, this release is production ready.

File: flathub.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches":["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: Why is this x86_64? It's a JavaScript application and I didn't see an explanation. (I could have missed it)

File: org.js.nuclear.Nuclear.json [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                },
+                {
+                    "type":"file",
+                    "url":"https://raw.githubusercontent.com/advaithm/nuclear/master/flatpak/org.js.nuclear.Nuclear.metainfo.xml",
+                    "sha256":"17f2d8221f3cfec579457c05a4a56b85dcff2dd44bc86d5a2bca85e088af7c39",
+                    "dest-filename":"org.js.nuclear.Nuclear.metainfo.xml"
+                },
+                {
+                  "type":"file",
+                  "url":"https://raw.githubusercontent.com/advaithm/nuclear/master/flatpak/org.js.nuclear.Nuclear.desktop",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:40]: if you are pulling directly from git like that, please do it on a specific commit. `master` will move and the content might change.

(same above and below)

File: org.js.nuclear.Nuclear.json [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+            ],
+        "build-commands": [
+                "mkdir -p /app/main /app/bin",
+                "cp -ra  * /app/main/",
+                "mkdir -p /app/share/metainfo/",
+                "mkdir -p /app/share/icons/hicolor/512x512/apps/",
+                "mkdir -p /app/share/applications/",
+                "cp -r org.js.nuclear.Nuclear.png /app/share/icons/hicolor/512x512/apps/",
+                "ls /app/share/icons/hicolor/512x512/apps/",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:64]: why a `ls` here?

File: org.js.nuclear.Nuclear.json [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+                "cp -r org.js.nuclear.Nuclear.metainfo.xml /app/share/metainfo/",
+                "cp -r org.js.nuclear.Nuclear.desktop /app/share/applications/",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:66]: Usually we use `install` instead of `cp`. It allow creating the directories if needed.

Also usually these go into the `post-install` section.

File: org.js.nuclear.Nuclear.json [hunk 9]
```diff
@@ -0,0 +1,71 @@
+{
+    "app-id": "org.js.nuclear.Nuclear",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "19.08",
+    "branch": "19.08",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:5]: why '19.08' branch?

File: flathub.json (hunk 5 above)
Comment (advaithm) [reply flathub.json:2]: Currently we only build for x86_64. We don't have plans to build for any other archs any time soon. Hence we would like to keep all packages relatively the same

File: org.js.nuclear.Nuclear.json (hunk 8 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json:66]: Will change

File: org.js.nuclear.Nuclear.json (hunk 6 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json:40]: We are going to be making a "mirror" repo which won't change unless the flatpak does too

File: org.js.nuclear.Nuclear.json (hunk 7 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json:64]: Sorry i used ls too get a better idea of the file structure

File: org.js.nuclear.Nuclear.json (hunk 6 above)
Comment (hfiguiere) [reply org.js.nuclear.Nuclear.json:40]: Either these file are in this repository (it's often the case for .desktop or .appdata.xml that aren't yet upstream) or you refer to them with a specific commit or tag.

File: org.js.nuclear.Nuclear.json (hunk 9 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json]: it is for the electron  base app

File: org.js.nuclear.Nuclear.json (hunk 6 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json:40]: is it possible to continue using the icons from the main repo

File: org.js.nuclear.Nuclear.json [hunk 10]
```diff
@@ -31,13 +31,13 @@
...
                     "type":"file",
-                    "url":"https://raw.githubusercontent.com/advaithm/nuclear/master/flatpak/org.js.nuclear.Nuclear.metainfo.xml",
+                    "url":"https://raw.githubusercontent.com/advaithm/flathub/nuclear/org.js.nuclear.Nuclear.metainfo.xml",
                     "sha256":"17f2d8221f3cfec579457c05a4a56b85dcff2dd44bc86d5a2bca85e088af7c39",
                     "dest-filename":"org.js.nuclear.Nuclear.metainfo.xml"
                 },
                 {
                   "type":"file",
-                  "url":"https://raw.githubusercontent.com/advaithm/nuclear/master/flatpak/org.js.nuclear.Nuclear.desktop",
+                  "url":"https://raw.githubusercontent.com/advaithm/flathub/nuclear/org.js.nuclear.Nuclear.desktop",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:40]: use the pah:
```json
"path": "org.js.nuclear.Nuclear.desktop"
```

File: org.js.nuclear.Nuclear.json [hunk 11]
```diff
@@ -31,13 +31,13 @@
                 },
                 {
                     "type":"file",
-                    "url":"https://raw.githubusercontent.com/advaithm/nuclear/master/flatpak/org.js.nuclear.Nuclear.metainfo.xml",
+                    "url":"https://raw.githubusercontent.com/advaithm/flathub/nuclear/org.js.nuclear.Nuclear.metainfo.xml",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:34]: use the pah:
```json
"path": "org.js.nuclear.Nuclear.metainfo.xml"
```

File: org.js.nuclear.Nuclear.json [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+                },
+                {
+                  "type":"file",
+                  "url":"https://raw.githubusercontent.com/advaithm/flathub/nuclear/org.js.nuclear.Nuclear.desktop",
+                  "sha256":"61256cb50cbfa54d93b9c344f4795f42f7dc73cd195be17915403d0ac281ce88",
+                  "dest-filename":"org.js.nuclear.Nuclear.desktop"
+                },
+                {
+                    "type":"file",
+                    "url":"https://github.com/nukeop/nuclear/raw/master/build/512.png",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:46]: copy this one too in the repository.

you can remove it when the new upstream has them.

File: org.js.nuclear.Nuclear.json (hunk 9 above)
Comment (hfiguiere) [reply org.js.nuclear.Nuclear.json:5]: I think what you want is `base-version` to be that. And `branch` left unset or to `stable`

File: org.js.nuclear.Nuclear.json (hunk 12 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json:45]: okay

File: org.js.nuclear.Nuclear.json [hunk 13]
```diff
@@ -2,9 +2,8 @@
     "app-id": "org.js.nuclear.Nuclear",
     "runtime": "org.freedesktop.Platform",
     "runtime-version": "19.08",
-    "branch": "19.08",
+    "branch": "0.6.3",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:5]: this is either not set (missing), or set to `stable`.

File: org.js.nuclear.Nuclear.json [hunk 14]
```diff
@@ -2,9 +2,8 @@
...
     "sdk": "org.freedesktop.Sdk",
-    "base": "io.atom.electron.BaseApp",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json]: why did you remove this if this is an electron app?

Comment (advaithm) [reply org.js.nuclear.Nuclear.json]: i removed it as it now use pre-compiled binaries  that contain all the required  things such as the electron binary. this makes the electron base app increase the size of the final product.

File: org.js.nuclear.Nuclear.json [hunk 15]
```diff
@@ -0,0 +1,67 @@
+{
+    "app-id": "org.js.nuclear.Nuclear",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "19.08",
+    "branch": "0.6.3",
+    "sdk": "org.freedesktop.Sdk",
+    "separate-locales": "false",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:7]: `separate-locales` is a boolean so it should be 
```json
 "separate-locales": false,
```

(no quotes around `false`)

flatpak-builder gives a warning about that.

File: org.js.nuclear.Nuclear.json (hunk 13 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json]: updated to stable although is there a pre release tag?

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

File: org.js.nuclear.Nuclear.json (hunk 15 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json:7]: oh okay i'll update it

**Comment by advaithm** [comment]
@hfiguiere i removed the  quotes

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

File: org.js.nuclear.Nuclear.json (hunk 14 above)
Comment (hfiguiere) [reply org.js.nuclear.Nuclear.json]: ah I see. I missed that. But maybe that how instead a real build could be done for non x86_64 architecture, in the future.

File: org.js.nuclear.Nuclear.json (hunk 11 above)
Comment (hfiguiere) [reply org.js.nuclear.Nuclear.json:33]: This is not what I meant. Remove `url`, path is enough. Also since the file is now local, you don't need the `sha256`

File: org.js.nuclear.Nuclear.json (hunk 10 above)
Comment (hfiguiere) [reply org.js.nuclear.Nuclear.json:39]: same here.

File: org.js.nuclear.Nuclear.json [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+                    "dest-filename": "run.sh",
+                    "commands": [ "/app/main/nuclear" ]
+                }
+
+            ],
+        "build-commands": [
+                "mkdir -p /app/main /app/bin /app/share/metainfo/  /app/share/applications/",
+                "cp -ra  * /app/main/",
+		        "mkdir -p /app/share/icons/hicolor/512x512/apps/",
+                "cp -r  org.js.nuclear.Nuclear.png /app/share/icons/hicolor/512x512/apps/",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:60]: like the rest use install.
Also `cp -r` is recursive which is pointless for a single file.

File: org.js.nuclear.Nuclear.json [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+                {
+                    "type": "script",
+                    "dest-filename": "run.sh",
+                    "commands": [ "/app/main/nuclear" ]
+                }
+
+            ],
+        "build-commands": [
+                "mkdir -p /app/main /app/bin /app/share/metainfo/  /app/share/applications/",
+                "cp -ra  * /app/main/",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:58]: `-r` is rundant with `-a`

Comment (advaithm) [reply org.js.nuclear.Nuclear.json:58]: okay will change

**Comment by advaithm** [comment]
@hfiguiere updated to use install and removed the a flag

File: org.js.nuclear.Nuclear.json (hunk 13 above)
Comment (hfiguiere) [reply org.js.nuclear.Nuclear.json]: This just means that it is the main branch for packages.

File: org.js.nuclear.Nuclear.json (hunk 11 above)
Comment (hfiguiere) [reply org.js.nuclear.Nuclear.json:33]: I said to remove `url` too.

File: org.js.nuclear.Nuclear.json [hunk 18]
```diff
@@ -31,19 +31,16 @@
                 {
                     "type":"file",
                     "url":"https://raw.githubusercontent.com/advaithm/flathub/nuclear/org.js.nuclear.Nuclear.metainfo.xml",
-                    "sha256":"17f2d8221f3cfec579457c05a4a56b85dcff2dd44bc86d5a2bca85e088af7c39",
                     "path": "org.js.nuclear.Nuclear.metainfo.xml"
                 },
                 {
                   "type":"file",
                   "url":"https://raw.githubusercontent.com/advaithm/flathub/nuclear/org.js.nuclear.Nuclear.desktop",
-                  "sha256":"61256cb50cbfa54d93b9c344f4795f42f7dc73cd195be17915403d0ac281ce88",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json]: I said to remove `url` too.

**Comment by advaithm** [comment]
i haven't removed the url as there are people who would like to build the flatpak on their own

File: org.js.nuclear.Nuclear.json [hunk 19]
```diff
@@ -31,19 +31,16 @@
...
                   "path": "org.js.nuclear.Nuclear.desktop"
                 },
                 {
                     "type":"file",
                     "url":"https://github.com/advaithm/flathub/raw/nuclear/512.png",
-                    "sha1":"f7568bc736304246d054faa7cf528c3e8bc0a6d5",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json]: And here I said to have the icon a local file like the .desktop and .appdata.xml added to this repository remove `url`, and instead specify `path`.

You can remove it when the upstream release has it.

File: org.js.nuclear.Nuclear.json (hunk 18 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json]: removed the url

File: org.js.nuclear.Nuclear.json (hunk 19 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json]: oh okay

File: org.js.nuclear.Nuclear.json [hunk 20]
```diff
@@ -30,18 +30,15 @@
...
                 {
                   "type":"file",
-                  "url":"https://raw.githubusercontent.com/advaithm/flathub/nuclear/org.js.nuclear.Nuclear.desktop",
                   "path": "org.js.nuclear.Nuclear.desktop"
                 },
                 {
                     "type":"file",
-                    "url":"https://github.com/advaithm/flathub/raw/nuclear/512.png",
-                    "dest-filename":"org.js.nuclear.Nuclear.png"
+                    "path":"org.js.nuclear.Nuclear.png"
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:41]: but the file in the repository is named "512.png" so this doesn't build.

Comment (advaithm) [reply org.js.nuclear.Nuclear.json:41]: I'll rename it

**Comment by hfiguiere** [comment]
> i haven't removed the url as there are people who would like to build the flatpak on their own. if it is a requirement i will remove it

I don't understand you here. It is all self contained. You don't need the URL at all. Anyone can build the flatpak checking this branch (or later the repository) out.

**Comment by advaithm** [comment]
oh okay yeah i didn't realize that path was a reference to a path outside the sandbox

**Comment by advaithm** [comment]
@hfiguiere i've updated the file name it should build now

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by hfiguiere** [comment]
I am of the opinion that this should be build from source using the base electron app instead of just using 'prebuilt'.

This would also allow considering to provide support for non-x86_64

**Comment by advaithm** [comment]
I actually originally wanted to build from source but due 2 factors I believe that this is better.
1. The build time from source on 24core vm was almost 1 hour.
2. I was not able to get the npm-cache working reliably. I have had the cache work one build and then immediately after if I ran again changing the name it would not use the cache.

In the future if I am able to get the app to build from source I will definitely be willing to shift to that until then this is the best solution I have.

**Comment by advaithm** [comment]
@hfiguiere after some research one of the packages used to build from source `lerna` requires internet access when boot strapping. there is really nothing that can be done to use a cache. i believe if we have to build for multiple architectures we will figure out a solution but until then this is the only one.

**Comment by advaithm** [comment]
@hfiguiere when you build for other architecters does the build run on a arm vm?

**Comment by nedrichards** [comment]
@advaithm The arm builds run on real arm server hardware from packet.net.

**Comment by advaithm** [comment]
@nedrichards thank you for the information

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
@hfiguiere  I have created a python script that will be used if nuclear starts supporting other archs. currently there are no plans.

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
@hfiguiere the python script will extract the appropriate binaries and install them if nuclear support other architectures. this will slove the issue of needing too use source.

File: org.js.nuclear.Nuclear.json [hunk 21]
```diff
@@ -0,0 +1,67 @@
...
+                  "type":"file",
+                  "path": "org.js.nuclear.Nuclear.desktop"
+                },
+                {
+                    "type":"file",
+                    "path":"org.js.nuclear.Nuclear.png"
+                },
+                {
+                    "type":"file",
+                    "url":"https://raw.githubusercontent.com/advaithm/flathub/nuclear/build_for_arch.py",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:46]: again, you want `path` not `url` since it is already here

Comment (advaithm) [reply org.js.nuclear.Nuclear.json:46]: yeah i forgot. fist time it ran i had the path thing. it failed.

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
@hfiguiere i have updated accordingly. does the python script satisfy the ability to build for multiple architectures.

**Comment by hfiguiere** [comment]
To be fair I don't see what it brings.

**Comment by advaithm** [comment]
@hfiguiere  my idea is to download all binaries. based on the cpu arch extract the appropriate binary and delete the others. this should allow building for multiple archs. this removes the need to build from source. again. it is simply not possible to build nuclear from source without network access due to the command `lerna bootstrap` downloading certain packages only required for the build process. the specific command `bootstrap` lacks the ability to use a pre-downloaded cache.

**Comment by hfiguiere** [comment]
This won't work. The build process doesn't have access to the network. So it can not download.

**Comment by advaithm** [comment]
@hfiguiere by download I mean the flatpak-builder stage will download them as sources

**Comment by advaithm** [comment]
also could someone explain to me how this will be published. the wiki doesn't really explain this section very well.

**Comment by hfiguiere** [comment]
Once this is approved by the maintainer, a new repository for this will be created with the content of that pull request and flathub will handle the build process.

**Comment by advaithm** [comment]
Thank you for the explanation. @hfiguiere. How long does this normally take?

File: org.js.nuclear.Nuclear.json [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+                {
+                    "type": "script",
+                    "dest-filename": "run.sh",
+                    "commands": [ "/app/main/nuclear" ]
+                }
+
+            ],
+        "build-commands": [
+                "python3 build_for_arch.py",
+		        "mkdir -p /app/share/icons/hicolor/512x512/apps/",
```
Comment (bilelmoussaoui) [inline org.js.nuclear.Nuclear.json:57]: You don't need to explicitly create a directory as you're using install later

File: org.js.nuclear.Nuclear.metainfo.xml [hunk 23]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bilelmoussaoui) [inline org.js.nuclear.Nuclear.metainfo.xml:1]: This should be upstreamed along with the desktop file & the icon.

File: org.js.nuclear.Nuclear.metainfo.xml [hunk 24]
```diff
@@ -0,0 +1,75 @@
...
+        <screenshot>
+            <caption>Equalizer View</caption>
+            <image>https://i.imgur.com/WreRL0w.png</image>
+        </screenshot>
+    </screenshots>
+    <url type="homepage">https://nuclear.js.org</url>
+    <project_group>Nuclear</project_group>
+    <provides>
+        <binary>Nuclear</binary>
+    </provides>
```
Comment (bilelmoussaoui) [inline org.js.nuclear.Nuclear.metainfo.xml:67]:
```suggestion
```
you don't need this

File: org.js.nuclear.Nuclear.json [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+    "branch": "stable",
+    "sdk": "org.freedesktop.Sdk",
+    "separate-locales": false,
+    "command": "run.sh",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--filesystem=host",
+	"--socket=pulseaudio"
```
Comment (bilelmoussaoui) [inline org.js.nuclear.Nuclear.json:14]: Nitpick: indentation

File: org.js.nuclear.Nuclear.json [hunk 26]
```diff
@@ -0,0 +1,65 @@
...
+    ],
+    "build-options" : {
+        "env": {
+            "NPM_CONFIG_LOGLEVEL": "info"
+        }
+    },
```
Comment (bilelmoussaoui) [inline org.js.nuclear.Nuclear.json:20]: You're not building it anymore, not needed

File: build_for_arch.py [hunk 27]
```diff
@@ -0,0 +1,12 @@
+import platform
```
Comment (bilelmoussaoui) [inline build_for_arch.py:1]: What's this script is used for ?

Comment (advaithm) [reply build_for_arch.py:1]: the script is used to build for multiple archs if and when nuclear decides to support them

File: org.js.nuclear.Nuclear.metainfo.xml (hunk 23 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.metainfo.xml:1]: the metainfo.xml and desktop file are not used by nuclear but are on the upstream repo should anyone want to inspect it. the icon has been taken from upstream.

File: org.js.nuclear.Nuclear.json (hunk 22 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json:57]: if I do not. the build fails because the directory does not exist @bilelmoussaoui

File: org.js.nuclear.Nuclear.json (hunk 25 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json]: i have fixed this

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

File: build_for_arch.py (hunk 27 above)
Comment (bilelmoussaoui) [reply build_for_arch.py:1]: No need for such abstraction to support something isn't being worked on.

Comment (hfiguiere) [reply build_for_arch.py:1]: and even them you can have modules specific to an architecture. So there would be no need for that.

File: org.js.nuclear.Nuclear.json [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+                    "type": "script",
+                    "dest-filename": "run.sh",
+                    "commands": [ "/app/main/nuclear" ]
+                }
+
+            ],
+        "build-commands": [
+                "python3 build_for_arch.py",
+		        "mkdir -p /app/share/icons/hicolor/512x512/apps/", 
+                "install  org.js.nuclear.Nuclear.png /app/share/icons/hicolor/512x512/apps/",
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.json:53]: install can create the directory for you. See the option `--target-directory`

File: org.js.nuclear.Nuclear.metainfo.xml [hunk 29]
```diff
@@ -0,0 +1,72 @@
...
+    <content_rating type="oars-1.1" />
+    <description>
+    <p>nuclear is a free music streaming program that pulls content from free sources all over the internet.</p>
+
+    <p>If you know mps-youtube, this is a similar music player but with a GUI.
+    It's also focusing more on audio. Imagine Spotify which you don't have to pay for and with a bigger library.</p>
+
+    <p>Features</p>
+    <ul>
+        <li>- Searching for and playing music from YouTube (including integration with playlists), Jamendo, and SoundCloud</li>
```
Comment (hfiguiere) [inline org.js.nuclear.Nuclear.metainfo.xml:17]: Since it's a `<li>` (list item) you shouldn't have a dash at the begining.

File: org.js.nuclear.Nuclear.json (hunk 28 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.json:53]: will update. thankyou for the information

File: org.js.nuclear.Nuclear.metainfo.xml (hunk 29 above)
Comment (advaithm) [reply org.js.nuclear.Nuclear.metainfo.xml:17]: okay, i will remove it.

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by advaithm** [comment]
I have reduced the file access permissions to only access to the home  dir. ideally this is the only area that users will use. should the need arise alternative file access then it can be looked into.

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

File: org.js.nuclear.Nuclear.metainfo.xml (hunk 23 above)
Comment (bilelmoussaoui) [reply org.js.nuclear.Nuclear.metainfo.xml:1]: Why not use the files from the upstream tarball then? I don't see a need to copy them over here.

Comment (advaithm) [reply org.js.nuclear.Nuclear.metainfo.xml:1]: the tarball does not have it.

Comment (bilelmoussaoui) [reply org.js.nuclear.Nuclear.metainfo.xml:1]: You can still use a source of type file and download them from the repo.

File: build_for_arch.py (hunk 27 above)
Comment (advaithm) [reply build_for_arch.py:1]: oh okay

**Comment by advaithm** [comment]
bot, build org.js.nuclear.Nuclear

**Comment by nedrichards** [comment]
bot, build org.js.nuclear.Nuclear
